plugin/metrics/prometheus/metricsstore/reader.go

```diff
@@ -63,24 +100,149 @@ func NewMetricsReader(logger *zap.Logger, hostPort string, connectTimeout time.D
 }
 
 // GetLatencies gets the latency metrics for the given set of latency query parameters.
-func (m *MetricsReader) GetLatencies(ctx context.Context, params *metricsstore.LatenciesQueryParameters) (*metrics.MetricFamily, error) {
-	// TODO: Implement me
-	return &metrics.MetricFamily{}, nil
+func (m *MetricsReader) GetLatencies(ctx context.Context, requestParams *metricsstore.LatenciesQueryParameters) (*metrics.MetricFamily, error) {
+	metricsParams := metricsQueryParams{
+		BaseQueryParameters: requestParams.BaseQueryParameters,
+		groupByHistBucket:   true,
+		metricName:          latenciesMetricName,
+		metricDesc:          fmt.Sprintf(latenciesMetricDesc, requestParams.Quantile),
+		buildPromQuery: func(p promQueryParams) string {
+			return fmt.Sprintf(
+				// Note: p.spanKindFilter can be ""; trailing commas are okay within a timeseries selection.
+				`histogram_quantile(%.2f, sum(latency_bucket{service_name =~ "%s", %s}) by (%s))`,
+				requestParams.Quantile,
+				p.serviceFilter,
+				p.spanKindFilter,
+				p.groupBy,
+			)
+		},
+	}
+	return m.executeQuery(ctx, metricsParams)
 }
 
 // GetCallRates gets the call rate metrics for the given set of call rate query parameters.
-func (m *MetricsReader) GetCallRates(ctx context.Context, params *metricsstore.CallRateQueryParameters) (*metrics.MetricFamily, error) {
-	// TODO: Implement me
-	return &metrics.MetricFamily{}, nil
+func (m *MetricsReader) GetCallRates(ctx context.Context, requestParams *metricsstore.CallRateQueryParameters) (*metrics.MetricFamily, error) {
+	metricsParams := metricsQueryParams{
+		BaseQueryParameters: requestParams.BaseQueryParameters,
+		metricName:          callsMetricName,
+		metricDesc:          callsMetricDesc,
+		buildPromQuery: func(p promQueryParams) string {
+			return fmt.Sprintf(
+				// Note: p.spanKindFilter can be ""; trailing commas are okay within a timeseries selection.
+				`sum(rate(calls_total{service_name =~ "%s", %s}[%s])) by (%s)`,
```

Comment:
> Why the sum of rates? Do you expect to have more than one time series for the given query?

Yup, unless if I'm mistaken, I think the rate will execute a per second delta over a given time window (e.g. `10m`) for each timeseries, and I don't think it will perform an aggregation across timeseries.

With that in mind, to help explain my intention, we'll start with a concrete example of the inner-most timeseries selector query (values simplified for readability), say: 
```
calls_total{service_name =~ "frontend", span_kind =~ "server"}
```

This can return multiple timeseries, for example:
- `calls_total{service_name = "frontend", operation="/foo", span_kind = "server"}`
- `calls_total{service_name = "frontend", operation="/bar", span_kind = "server"}`
- etc.

and say if we want the query to return the call rate grouped by `service_name` **only**, then we'd need to somehow aggregate all the above timeseries into one that looks like:
- `call_rate{service_name = "fronted"}` 

> If so, would the rate still be accurate?

I guess what I have in mind is to give some sort of indication, at a service-level (or operation-level per service), what the QPS is. 
In the above example, for a given point in time, if operation `/foo` has a rate of 1 request/s and `/bar` of 2 request/s, then I think what we'd like to see is a data point of 3 request/s, which is why I used `sum` on each of the rates.

> Should I summon a Prometheus expert to review this?

If it's not too much trouble, I'd appreciate that as I'm very much a novice with PromQL, and any tips to make the query more correct or performant would be great.

Reply:
Yes, rate() before sum() is always the right thing to do. See https://promcon.io/2017-munich/slides/best-practices-and-beastly-pitfalls.pdf for a nice explanation :)

Reply:
Thanks @simonpasquier :) 

> rate() before sum() is always the right thing to do

Does this mean we should do `rate(sum(...))` or `sum(rate(...))`?

Reply:
Oh, I just saw in the link that it should always be `sum(rate(...))`, so looks like what I've got it okay. I also saw this: https://www.robustperception.io/rate-then-sum-never-sum-then-rate. 

Anything else that you can see in these queries that needs addressing?

Reply:
Thanks @simonpasquier!